crates/language/src/lib.rs

```diff
@@ -97,6 +97,56 @@ macro_rules! impl_lang_expando {
   };
 }
 
+/// Implements the `ALIAS` associated constant for the given lang, which is
+/// then used to define the `alias` const fn and a `Deserialize` impl.
+/// Also generates as convenience conversions between the lang types
+/// and `SupportedType`.
+macro_rules! impl_aliases {
+  ($($lang:ident => $as:expr),* $(,)?) => {
+    $(
+      impl $lang {
```

Comment:
I would suggest moving this macro into a standalone macro `impl_alias`. And then call the `impl_alias!` here

Reply:
What would the difference be?

Reply:
It will be a cleaner separate between defining one alias and defining multiple aliases. Macro is harder to understand and separating them into smaller pieces improve reading experience.

Reply:
It'd end up looking exactly the same except for the `$(...),*`. Having all the variants listed in one place allows the `alias` function to be implemented without adding boilerplate as well. Though separating out `alias` and including other boilerplatey functions (e.g. `all_langs`) in the macro could make sense. IMO the fewer times every language needs to be listed out, the better.

Reply:
I can address it later. Not blocking now.